### PR TITLE
Fix missing overload variable in Rocksdb ivf demo

### PR DESCRIPTION
**Pull request by Warmchay**
**Bugs:** 
When following rocksdb_ivf demo to build executable file, its output as:
```bash
faiss/demos/rocksdb_ivf/RocksDBInvertedLists.h:52:35: error: 'faiss::InvertedListsIterator* faiss_rocksdb::RocksDBInvertedLists::get_iterator(size_t) const' marked 'override', but does not override
   52 |     faiss::InvertedListsIterator* get_iterator(size_t list_no) const override;
      |                                   ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/demo_rocksdb_ivf.dir/build.make:90: CMakeFiles/demo_rocksdb_ivf.dir/RocksDBInvertedLists.cpp.o] Error 1
```

**Solution:**
Add relevant variable `void* inverted_list_contex` corresponding `get_iterator`'s base virtual function.

### CODE REVIEW

**Comment by mdouze** [comment]
@algoriddle would you mind taking a look?

**Comment by mdouze** [comment]
It would be useful to compiling this in CI so that we catch these kinds of API pdates.

File: demos/rocksdb_ivf/RocksDBInvertedLists.cpp [hunk 1]
```diff
@@ -101,7 +101,8 @@ void RocksDBInvertedLists::resize(size_t /*list_no*/, size_t /*new_size*/) {
 }
 
 InvertedListsIterator* RocksDBInvertedLists::get_iterator(
-        size_t list_no) const {
+        size_t list_no,
+        void* inverted_list_contex) const {
```
Comment (algoriddle) [inline demos/rocksdb_ivf/RocksDBInvertedLists.cpp:105]: thank you for this PR, can you correct the typo please (also below)? contex -> context

Comment (Warmchay) [reply demos/rocksdb_ivf/RocksDBInvertedLists.cpp]: No problem~

**Pull request closed without merging**